miprometheus/workers/offline_trainer.py

```diff
@@ -288,6 +291,11 @@ def run_experiment(self):
                 # Apply curriculum learning - change some of the Problem parameters
                 self.curric_done = self.training_problem.curriculum_learning_update_params(episode)
 
+                # III. The episodes number limit has been reached. (2nd check)
+                if episode+1 >= self.episode_limit:
```

Comment:
Why a second check?

Reply:
Required to break outer (epoch) loop when episode limit reached.